### ISSUE TITLE
adding edfWrite?

### ISSUE THREAD

**Issue by bnicenboim** [Pisca46/edfReader#4]
Hi, Thanks for the useful package!
I know it's edf*Reader*, but it would be nice if the package would allow to write back edf files after modifying the edf objects that edfReader produces.

**Comment by Pisca46** [comment]
Hi,

First, I am sorry for this extreme late response. 

I agree that it would be nice to extent the edfReader package to an edfReaderWriter package.

However I don’t have the time to write it.

Being retired, I am too often claimed by my (grand)kids and I am currently working at another package more in line which my actual research interest.

Nevertheless, if I could be of any assistant for somebody else who wants to do the job, I will be glad to do so.

Best regards

Pisca46

**Comment by a-hurst** [comment]
@Pisca46 I recently wrote a pure-Python EDF+ writer for getting EDFs out of Python and into R, and I feel pretty confident the code wouldn't be too difficult to bring over to R, so I'd be willing to take a crack at a pull request if you're interested!

@bnicenboim What were you imagining the API to be like for this function? I'm imagining a function that takes a dataframe and uses the data and column names to generate the file, with extra arguments for annotations and for specifying file metadata (could be an edfHeader object or a list). That would mean you could import (or generate) EEG data from other formats/packages in R and still be able to write them out to EDF.

**Comment by bnicenboim** [comment]
yeah that would be great, I guess it should take the metadata as list, and probably a matrix (to speed things up) with the eeg data. I'm actually storing the EEG data in `data.table`s, since they are faster than data frames or tibbles. So a data table or a matrix would be great from my side.

**Comment by Pisca46** [comment]
Hi,


First, I am sorry again for my extreme late response. 

Thank you for your offer “to take a crack at a pull request “.

However, as explained in my mail to @bnicenboim <https://github.com/bnicenboim>  dd Feb 26, 2019, I don’t have the time to work on an edfWriter package. Moreover such a package is not in my personal interest too.

Best regards

Piska46


PS

I am not sure whether this is of intrest for you, but storing EDF signals in a single data frame may be be cause problems as each EDF signal may have a different sample rate.

**Comment by a-hurst** [comment]
@Pisca46 Thank you for your response! I fully understand no longer having time or interest in maintaining a package you no longer need yourself.

To make sure your hard work on edfReader is preserved for future versions of R, would you be open to transferring maintenance of the package to someone else? All that's required of you is sending a one-sentence email to CRAN-submissions@R-project.org (from the email address you use for the package), containing the name and email address of the person you're transferring maintainership to. CRAN will take care of the rest.

If either @craddm (of the eegUtils project) or @bnicenboim are interested I'll defer to them, but if not I'd be happy to do it myself. I've taken over maintenance of another abandoned CRAN pacakge before so I know what's involved!

**Comment by bnicenboim** [comment]
@a-hurst, I'll be glad if you take over!

**Comment by Pisca46** [comment]
@a-hurst <https://github.com/a-hurst>,  First my best wishes for 2021 and, again, my apologies for my late response.

Thank you for your offer to maintain the edfReader package. However I have  trouble in doing it myself for the next few years. Its a full implementation (with no outstanding requests for corrections or new features) and also R is quite stable. 

Best regards,

Piska46
